evalml/tests/automl_tests/dask_tests/test_automl_dask.py

```diff
@@ -22,9 +22,7 @@ def sequential_engine():
 
 @pytest.fixture(scope="module")
 def cluster():
-    dask_cluster = LocalCluster(
-        n_workers=1, threads_per_worker=2, dashboard_address=None
-    )
+    dask_cluster = LocalCluster(n_workers=1, dashboard_address=None)
```

Comment:
Great find! I guess the thinking is that the test relies on `TestPipelineFast` finishing before `TestPipelineWithFitError` but since we set `threads_per_worker` to 2 and there are three pipelines, we can't guarantee that all three pipelines start evaluating at the same time.

To be extra safe, can you verify what the value of `CPU_COUNT` (referenced in the dask [source code](https://github.com/dask/distributed/blob/main/distributed/deploy/local.py#L195)) in our gh workers is? 

One way to do that may be to add a test that fails like this:

```python
from dask.system import CPU_COUNT
assert CPU_COUNT == None
```

The benefit of doing this is verifying that threads_per_worker gets set to 16 as opposed to something else. Like you said, even going with 8 can cause failures.

Reply:
FWIW, I think I'm good with merging this in and seeing how it goes. I just want to know what `CPU_COUNT` is on gh so we know what to expect with regards to future flakes. If this fails, we can try to rewrite the test to not rely a specific pipeline evaluation order?

Reply:
It's 16!

Reply:
In that case, let's get this baby merged and see what happens 😂

Reply:
It's 2 lol
![image](https://user-images.githubusercontent.com/22552445/124830796-43ddd300-df2f-11eb-91a0-7eb568aff8b0.png)

Reply:
😨  😂  🙈

Reply:
In that case, how about we change the test? The test is supposed to verify that all computations stop after we hit an exception with the `raise_error_callback`.

I think this line that we already have in the test (`assert len(automl.full_rankings) < len(pipelines)`) is sufficient. Whether or not `TestPipelineFast` actually finishes is not really relevant.

What do you think? FYI @chukarsten

Reply:
I think that sounds good to me. I agree with not requiring the fast pipeline be in the results